src/dask_awkward/io.py

```diff
@@ -61,15 +65,109 @@ def _from_json_bytes(source) -> ak.Array:
     )
 
 
+def derive_json_meta(
+    source: str,
+    compression: str | None = "infer",
+    sample_rows: int = 5,
+    bytechunks: str | int = "8 KiB",
+    force_by_lines: bool = False,
+) -> ak.Array:
+    if compression == "infer":
+        compression = infer_compression(source)
+
+    bytechunks = parse_bytes(bytechunks)
+
+    # when the data is uncompressed we read `bytechunks` number of
+    # bytes then split on a newline bytes, and use the first
+    # `sample_rows` number of lines.
+    if compression is None and not force_by_lines:
+        try:
+            with fsspec.open(source, mode="rb") as f:
+                byteread = f.read(bytechunks)
+                lines = [
+                    json.loads(ln) for ln in byteread.split(b"\n")[:sample_rows] if ln
+                ]
+                return ak.Array(ak.from_iter(lines).layout.typetracer.forget_length())
+        except ValueError:
+            # we'll get a ValueError if we can't decode the JSON from
+            # the bytes that we grabbed.
+            warnings.warn(
+                f"Couldn't determine metadata from reading first {bytechunks} "
+                f"of the dataset; will read the first {sample_rows} instead. "
+                "Try increasing the value of `bytechunks` to remove this warning."
+            )
+
+    # for compressed data (or if explicitly asked for with
+    # force_by_lines set to True) we read the first `sample_rows`
+    # number of rows after opening the compressed file.
+    with fsspec.open(source, mode="rt", compression=compression) as f:
+        lines = []
+        for i, line in enumerate(f):
+            lines.append(json.loads(line))
+            if i >= sample_rows:
+                break
+        return ak.Array(ak.from_iter(lines).layout.typetracer.forget_length())
```

Comment:
@martindurant I'm curious what your thoughts are on this `derive_json_meta` function (called in the case of `from_json(..., meta=None)` (which is default)